public/jira/derived/work-status/work-status.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * This module is repsonsible for determining the correct workType ("design", "dev", "qa", "uat")
+ * and statusType ("qa", "uat", "todo", "done", "blocked") for an issue.
+ */
+export const workType = ["design", "dev", "qa", "uat"] as const;
+export const workTypes = workType;
+
+export const workflowHappyPath = ["todo", "design", "dev", "qa", "uat", "done"];
+export const workflowUnhappyStatuses = ["blocked"];
+
+const inQAStatus = { qa: true, "in qa": true, "qa complete": true };
+const inPartnerReviewStatus = { "partner review": true, uat: true };
+const inDoneStatus = { done: true, cancelled: true };
+const blockedStatus = { blocked: true, delayed: true };
+
+export const inPartnerReviewStatuses = Object.keys(inPartnerReviewStatus);
+export const inIdeaStatus = { idea: true, "to do": true, open: true };
+export const inIdeaStatuses = Object.keys(inIdeaStatus);
+
+type WorkType = (typeof workType)[number];
+
+type Status =
```

Comment:
nitpick: Not related to the scope of this PR. But I realized that the pattern in this repo is to include the types as part of the ts files.
What do you think about extracting them in their own `interface` files. It will improve readability + we can move any common interfaces to the `interface` file at the root of the `jira` folder or at the folder level that makes more sense

Reply:
I prefer to have my types as close to the things they are typing as possible (and prefer to collocated types next to features then the other way around) and If it turns out it's needed in other places then consider lifting up. So I'm not against the idea, I just want to make sure other places need them

Reply:
I see your point too in keeping things in the same file, each approach has its own trade offs.
I am fine either way too. Let's keep them then as it is.